### PR TITLE
support memory swap by CRI

### PR DESCRIPTION
**Pull request by wangforthinker**
Signed-off-by: allen.wang <allen.wq@alipay.com>

### Ⅰ. Describe what this PR did
Support to set memory swap by CRI interface. If set key resources.memory_swap in annotation, pouch will set it to container config.

### Ⅱ. Does this pull request fix one issue?

Fix https://github.com/alibaba/pouch/issues/2698.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
add testcase.

### Ⅳ. Describe how to verify it
do it by testcase.

### CODE REVIEW

File: cri/v1alpha2/cri_utils.go [hunk 1]
```diff
@@ -1258,3 +1265,21 @@ func toCNIPortMappings(criPortMappings []*runtime.PortMapping) []ocicni.PortMapp
...
 	return portMappings
 }
+
+// applyContainerConfigByAnnotation updates pouch container config according to annotation.
+func applyContainerConfigByAnnotation(annotations map[string]string, podSandboxID string, config *apitypes.ContainerConfig, hc *apitypes.HostConfig) error {
+	if len(annotations) == 0 {
+		return nil
+	}
+
+	if memorySwap, ok := annotations["resources.memory_swap"]; ok {
```
[the remark this answers is not in the document]
Comment (wangforthinker) [reply cri/v1alpha2/cri_utils.go]: Fix it!

File: cri/v1alpha2/cri_utils_test.go [hunk 2]
```diff
@@ -1854,3 +1855,50 @@ func Test_toCNIPortMappings(t *testing.T) {
 		})
 	}
 }
+
+// CRI test: apply container config by annotation
+func Test_applyContainerConfigByAnnotation(t *testing.T) {
```
Comment (zhuangqh) [inline cri/v1alpha2/cri_utils_test.go]: The name of this function is `applyContainerConfigByAnnotation`. It means there would be other annotation follows, not just `resource.memory_swap`. 
I think we should make the test more generally for the coming annotation, change the parameters to `annotation map[string]string`

File: cri/v1alpha2/cri_utils_test.go [hunk 3]
```diff
@@ -1854,3 +1855,58 @@ func Test_toCNIPortMappings(t *testing.T) {
...
+	tests := []struct {
+		name       string
+		annotation map[string]string
+		checkFn    func(config *apitypes.ContainerConfig, hc *apitypes.HostConfig) bool
```
Comment (zhuangqh) [inline cri/v1alpha2/cri_utils_test.go]: cool!👍

**Comment by wangforthinker** [comment]
ping @zhuangqh

File: cri/v1alpha2/cri_utils.go [hunk 4]
```diff
@@ -1258,3 +1265,21 @@ func toCNIPortMappings(criPortMappings []*runtime.PortMapping) []ocicni.PortMapp
 	}
 	return portMappings
 }
+
+// applyContainerConfigByAnnotation updates pouch container config according to annotation.
+func applyContainerConfigByAnnotation(annotations map[string]string, podSandboxID string, config *apitypes.ContainerConfig, hc *apitypes.HostConfig) error {
```
Comment (fuweid) [inline cri/v1alpha2/cri_utils.go]: both the `podSandboxID` and `config` are useless right now, how about to remove it?

Comment (zhuangqh) [reply cri/v1alpha2/cri_utils.go]: we may apply annotations to the `config` or `hc`

Comment (wangforthinker) [reply cri/v1alpha2/cri_utils.go]: @fuweid @zhuangqh  Have removed podSandboxID.

**Pull request merged**